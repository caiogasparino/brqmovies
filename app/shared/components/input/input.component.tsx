import { MaterialCommunityIcons } from "@expo/vector-icons";
import React from "react";
import { TextInput } from "react-native-paper";
import { Input } from "./input.styles";

interface InputProps extends React.ComponentProps<typeof Input> {
    leftIcon?: React.ReactNode;
    rightIcon?: React.ReactNode;
    leftIconName?: keyof typeof MaterialCommunityIcons.glyphMap;
    rightIconName?: keyof typeof MaterialCommunityIcons.glyphMap;
    colorIcon?: string;
    secureTextEntry?: boolean;
    value?: string;
    testID?: string;
    onPress?: () => void;
}

const InputCustom: React.FC<InputProps> = ({
    placeholder,
    leftIcon,
    rightIcon,
    leftIconName,
    rightIconName,
    secureTextEntry,
    colorIcon,
    value,
    onPress,
    onChangeText,
    testID,
    ...props
}) => {
    const renderLeftIcon = () => {
        if (leftIcon) {
            return (
                <TextInput.Icon
                    testID={`${testID}-left-icon`}
                    icon={
                        leftIconName as keyof typeof MaterialCommunityIcons.glyphMap
                    }
                    color={colorIcon}
                />
            );
        }
        return null;
    };

    const renderRightIcon = () => {
        if (rightIcon) {
            return (
                <TextInput.Icon
                    testID={`${testID}-right-icon`}
                    icon={
                        rightIconName as keyof typeof MaterialCommunityIcons.glyphMap
                    }
                    color={colorIcon}
                    onPress={onPress}
                />
            );
        }
        return null;
    };

    return (
        <Input
            testID={testID}
            placeholder={placeholder}
            left={renderLeftIcon()}
            right={renderRightIcon()}
            secureTextEntry={secureTextEntry}
            value={value}
            onChangeText={onChangeText}
            {...props}
        />
    );
};

export { InputCustom };
