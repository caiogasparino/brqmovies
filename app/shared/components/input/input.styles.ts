import styled, { css } from "styled-components/native";
import { TextInput } from "react-native-paper";

export type INPUT_VARIANT = "PRIMARY" | "SECONDARY" | "TERTIARY";

type InputProps = {
    variant: INPUT_VARIANT;
};

export const Input = styled(TextInput)<InputProps>`
    border-radius: 4px;
    margin-bottom: 50px;
    ${({ variant }) =>
        variant === "PRIMARY" &&
        css`
            background-color: ${({ theme }) => theme.COLORS.BASE[400]};
            color: ${({ theme }) => theme.COLORS.BASE[100]};
            font-family: ${({ theme }) => theme.FONTS.PRIMARY[400]};
            font-size: ${({ theme }) => theme.METRICS.FONT_SIZE.x14}px;
            border-bottom: 2px solid ${({ theme }) => theme.COLORS.BASE[100]};
        `}

    ${({ variant }) =>
        variant === "SECONDARY" &&
        css`
            background-color: ${({ theme }) => theme.COLORS.NEUTRAL[100]};
            color: ${({ theme }) => theme.COLORS.BASE[100]};
            font-family: ${({ theme }) => theme.FONTS.PRIMARY[400]};
            font-size: ${({ theme }) => theme.METRICS.FONT_SIZE.x14}px;
            border: 1px solid ${({ theme }) => theme.COLORS.DARK[100]};
        `}

    ${({ variant }) =>
        variant === "TERTIARY" &&
        css`
            background-color: ${({ theme }) => theme.COLORS.DARK[100]};
            color: ${({ theme }) => theme.COLORS.NEUTRAL[100]};
            font-family: ${({ theme }) => theme.FONTS.PRIMARY[400]};
            font-size: ${({ theme }) => theme.METRICS.FONT_SIZE.x14}px;
            border: 1px solid ${({ theme }) => theme.COLORS.NEUTRAL[200]};
        `}
`;
