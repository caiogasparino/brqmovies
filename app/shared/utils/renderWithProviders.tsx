import { render, RenderAPI } from "@testing-library/react-native";
import React from "react";
import { SafeAreaProvider } from "react-native-safe-area-context";

import { NavigationContainer } from "@react-navigation/native";
import { THEME_LIGHT } from "app/shared/design/themes/theme-light";
import { ThemeProvider } from "styled-components";

const renderWithProviders = (component: React.ReactElement): RenderAPI => {
    return render(
        <NavigationContainer>
            <SafeAreaProvider>
                <ThemeProvider theme={THEME_LIGHT}>{component}</ThemeProvider>
            </SafeAreaProvider>
        </NavigationContainer>,
    );
};

export { renderWithProviders };
