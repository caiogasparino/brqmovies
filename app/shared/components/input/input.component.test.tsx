import { renderWithProviders } from "app/shared/utils/renderWithProviders";
import { InputCustom } from "./input.component";

describe("InputCustom component", () => {
    it("matches snapshot", () => {
        const { toJSON } = renderWithProviders(
            <InputCustom placeholder="Snapshot Test" variant={"PRIMARY"} />,
        );
        expect(toJSON()).toMatchSnapshot();
    });
});
