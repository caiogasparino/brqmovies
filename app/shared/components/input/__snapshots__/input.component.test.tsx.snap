// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCustom component matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
